astronomer/providers/cncf/kubernetes/operators/kubernetes_pod.py

```diff
@@ -28,8 +29,9 @@ class KubernetesPodOperatorAsync(KubernetesPodOperator):
     :param poll_interval: interval in seconds to sleep between checking pod status
     """
 
-    def __init__(self, *, poll_interval: int = 5, **kwargs: Any):
+    def __init__(self, *, poll_interval: int = 5, logging_interval: Optional[int] = None, **kwargs: Any):
```

Comment:
Can we add this to docstring above on L29 and update the warning on L25 too please.

And looks like tests are passing 🎉 

cc @pankajastro / @pankajkoti  / @phanikumv can one of you test this on our deployment to see the logging behaviour?

Reply:
Looks fine to me. I have set the interval to 10 seconds, but across 3 separate runs it resumed to gather the logs with an interval of 15-17 seconds and then got deferred again.

Reply:
https://user-images.githubusercontent.com/10206082/168273415-f5194af5-0bcd-4959-8e14-1ed62fb32f1c.mp4

@dstandish Attaching a screen-recording for the observed run

Reply:
Also, @kaxil  has the following points:

1. Should we disable the mini scheduler once we go-back to the worker just for fetching logs?
2. We probably want to add a log line saying - "Task is not completed, we are just fetching logs periodically" etc

Reply:
> Can we add this to docstring above on L29 and update the warning on L25 too please.

done

Reply:
> * Should we disable the mini scheduler once we go-back to the worker just for fetching logs?
> * We probably want to add a log line saying - "Task is not completed, we are just fetching logs periodically" etc

not familiar with the mini scheduler. @kaxil can you point me to any info / docs / code?  

re logging, one thing though, when we resume, we don't necessarily know that the task isn't done.  it could have finished after trigger fired "resume" event but before task is picked up.  or it could finish during logs read.  we get the container status immediately after fetching the latest logs, and at that time if it's still running we defer again.  but i suppose we could add a line that explains why we're deferring again...

Reply:
> but i suppose we could add a line that explains why we're deferring again..

done this now

Reply:
@dstandish Mini-scheduler is the scheduling that happens on workers for optimizations controlled by [[scheduler] schedule_after_task_execution](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#schedule-after-task-execution)

Reply:
and also mentioned in https://airflow.apache.org/docs/apache-airflow/stable/concepts/scheduler.html?highlight=schedule_after_task_execution#scheduler-configuration-options

Reply:
One Mypy failure too::

```
astronomer/providers/cncf/kubernetes/operators/kubernetes_pod.py: note: In member "defer" of class "KubernetesPodOperatorAsync":
astronomer/providers/cncf/kubernetes/operators/kubernetes_pod.py:51: error:
Function is missing a type annotation  [no-untyped-def]
        def defer(self, last_log_time=None, **kwargs):
        ^
Found 1 error in 1 file (checked 136 source files)

Exited with code exit status 1

```